examples/js/fortunewhale.js

```diff
@@ -0,0 +1,168 @@
+        ]
+    });
+
+    return fw.Workflow({
+        output: generateFortune,
+        tasks: [
+            generateFortune,
+            whaleSays
+        ]
+    })
+};
+
+// Simple dependencies with task IDs in the requires
+module.exports = function () {
+    const generateFortune = fw.Task({
+        id: "generateFortune",
+        run: "fortune"
+    });
+
+    const whaleSays = fw.task({
+        run: "whalesay",
+        inputs: {
+            "body": "{ output('generateFortune') }"
+        },
+        requires: [
+            generateFortune.id
+        ]
+    });
+
+    return fw.Workflow({
+        output: generateFortune,
+        tasks: [
+            generateFortune,
+            whaleSays
+        ]
+    })
+};
+
+// Referencing a task in an expression without an ID.
+module.exports = function () {
+    const generateFortune = fw.Task({
+        run: "fortune"
+    });
+
+    const whaleSays = fw.task({
+        run: "whalesay",
+        inputs: {
+            "body": `{ output('${generateFortune.id}') }`,  // or:
+            "body2": output(generateFortune.id),            // or:
+            "body3": output(generateFortune),
+        },
+        requires: [
+            generateFortune.id
+        ]
+    });
+
+    return fw.Workflow({
+        output: generateFortune,
+        tasks: [
+            generateFortune,
+            whaleSays
+        ]
+    })
+};
+
+// Sequence
+module.exports = function () {
+    return fw.sequence([
+        fw.task({
+            id: "generateFortune", // optional ID
+            run: "fortune"
+        }),
+        fw.sleep(1000),
+        fw.task({
+            run: "whalesay",
+            inputs: {
+                "body": "{ output('generateFortune') }"
+            }
+        })
+    ])
+};
+
+// Expression using helper functions
+module.exports = function () {
+    return fw.sequence([
+        fw.task({
+            id: "generateFortune", // optional ID
+            run: "fortune"
+        }),
+        fw.sleep(1000),
+        fw.task({
+            run: "whalesay",
+            inputs: {
+                "body": "{ output('generateFortune') }"
+            }
+        })
+    ])
+};
+
+// Expression with a function
+module.exports = function () {
+    return fw.sequence([
+        fw.task({
+            id: "generateFortune", // <- ID is optional, but needed to reference the function
+            run: "fortune"
+        }),
+        fw.sleep(1000), // <- how to specify ID with these built-in functions?
+        fw.task({
+            run: "whalesay",
+            inputs: {
+                "body": function (scope) {
+                    return scope.tasks["generateFortune"].output
+                }
+            }
+        })
+    ])
+};
+
+// Task with a function
+module.exports = function () {
+    return fw.sequence([
+        fw.task({
+            id: "generateFortune", // <- ID is optional, but needed to reference the function
+            run: function (inputs) {
```

Comment:
this is interesting, when is this function run?

Reply:
If run is a function it gets executed as an internal JavaScript function. It is identical to defining a `js` task: https://github.com/fission/fission-workflows/pull/200/files#diff-328c47b638336c765a36ff719ccdae57R78

Reply:
Neat!